src/Uno.UI/UI/Xaml/DependencyObjectStore.cs

```diff
@@ -1657,6 +1679,40 @@ internal IEnumerable<ResourceBinding> GetResourceBindingsForProperty(DependencyP
 			=> _resourceBindings?.GetBindingsForProperty(dependencyProperty)
 				?? Enumerable.Empty<ResourceBinding>();
 
+		/// <summary>
+		/// Enables hard references for internal fields for faster access
+		/// </summary>
+		/// <remarks>
+		/// Calling this method may cause memory leaks and must be used along with <see cref="DisableHardReferences"/>.
+		/// The use case is for FrameworkElement instances that can be Loaded and Unloaded, and for which the Unloaded
+		/// state is ensured to happen in such a way that memory leaks cannot happen.
+		/// </remarks>
+		internal void EnableHardReferences()
```

Comment:
Given that it depends on the feature flag, maybe `TryEnableHardReferences()`? (And if so, the same below)